### ISSUE TITLE
Allow sorting observations by biogeographical province 

### ISSUE THREAD

**Issue by mikkohei13** [luomus/laji#648]
When user sorts observation search result table based on biogeographical provice, use gathering.interpretations.biogeographicalProvince for sorting. (Not it uses gathering.interpretations.biogeographicalProvinceDisplayname, which makes the sort order alphabetic.)

**Comment by rpulkka** [comment]
@mikkohei13 Does this look correct: https://648.dev.laji.fi/observation/list?time=2024-01-01%2F

**Comment by mikkohei13** [comment]
No. Using this data for testing and showing havainnon lisätiedot, sorting by biogeographical province should sort the numbers in order.

https://648.dev.laji.fi/observation/list?time=2024-12-12%2F2024-12-12

![Image](https://github.com/user-attachments/assets/97b530c3-3e58-4539-9711-dcde22a5dfc5)

**Comment by mikkohei13** [comment]
Does this mean sorting information is not available through API?

**Comment by rpulkka** [comment]
@mikkohei13 I think yes. The query that happens when sorting you list is:

https://648.dev.laji.fi/api/warehouse/query/list?time=2024-12-12/2024-12-12&aggregateBy=unit.interpretations.recordQuality,document.linkings.collectionQuality,unit.linkings.taxon.taxonomicOrder,unit,unit.abundanceString,gathering.displayDateTime,gathering.interpretations.countryDisplayname,gathering.interpretations.biogeographicalProvinceDisplayname,gathering.locality,document.collectionId,document.documentId,gathering.team,unit.unitId,document.documentId&selected=unit.interpretations.recordQuality,document.linkings.collectionQuality,unit.linkings.taxon.taxonomicOrder,unit,unit.abundanceString,gathering.displayDateTime,gathering.interpretations.countryDisplayname,gathering.interpretations.biogeographicalProvinceDisplayname,gathering.locality,document.collectionId,document.documentId,gathering.team,unit.unitId,document.documentId&orderBy=gathering.interpretations.biogeographicalProvince%20ASC&cache=false&page=1&pageSize=100

Which includes `orderBy=gathering.interpretations.biogeographicalProvince` and produces the wrong result.

@esko-piirainen Could you take a look at this?

**Comment by esko-piirainen** [comment]
Unfortunately api does not support this after all -- the sorting is not done in ID order but rather in internal random "key order"

Should try to remember to fix it when doing  dw vertica schema v2

**Comment by esko-piirainen** [comment]
Unfortunately api does not support this after all -- the sorting is not done in ID order but rather in internal random "key order"

Should try to remember to fix it when doing dw vertica schema v2
